### DISCUSSION TITLE
Shortcut to a website - open the site with JavaScript enabled

### DISCUSSION THREAD

**Discussion by francwalter** [Waboodoo/HTTP-Shortcuts#447] [Q&A] [answered]
Hallo

I created a shortcut on my Desktop for my Tasmota, which is connected to my smart meter (de: "Stromzähler").
When I launch the site (192.168.0.213) at the spot, where the interesting information about current values should be placed, I get the message:

`To use Tasmota, please enable JavaScript`

Is it in any way possible to use HTTP-Shortcuts to show websites by using their JS, with "Javascript enabled"?
I didnt find any settings about it, neither here in Discussions.

I know that I can place Javascript myself in the settings of a shortcut, but is it not possible to use the site's JS?

One nice thing about HTTP-Shortcuts is that it is so fast. I could open my Tasmota Smartmeter with my browser (Firefox/Chrome/etc.) but that is so slow.

Thank for help, 
frank

--
Android 9 with Xiaomi Mi A1

**Comment by Waboodoo** [comment, marked as answer]
Hi,

This is currently not possible, as the app doesn't behave like a normal browser. It essentially just makes an HTTP request and then displays the HTML from the response. So basically, what's being displayed there isn't a website, it's a static snapshot of the website. I have considered allowing to enable JavaScript within that snapshot, but there would be no guarantee that it would actually work that code wouldn't be running inside a regular browser. So this could be an "experimental" feature at best.

My guess is that one of the main reasons why the HTTP Shortcuts app appears to load faster than a regular browser is *because* it isn't a browser, so it saves time by not loading any JavaScript for example.

So for now at least, there's no solution for making this work in the app. You could try creating a "Browser Shortcut" instead of a "Regular HTTP Shortcut" and switching its browser to "Custom Tab". That would essentially open a sort-of lite version of a browser, so maybe that would get you the desired performance.

**Reply by francwalter** [reply]
I was afraid that this would be like that, but hope dies last 😄 
Thank for the thorough answer. A browser shortcut was what I used to do but as you say, HTTP-Shortcuts is much faster 😄 
But could you implement, if not too much work, such a experimental feature? I imagine that it could work in some sites...
Thank anyway!!!

**Reply by Waboodoo** [reply]
I played around with it for a while and found that it works more reliably than I had feared, so I decided to add such an experimental option with the next version, which will probably be released within the next few weeks.

**Reply by francwalter** [reply]
That are good news, thank a lot! I will test it when published :)

**Reply by Waboodoo** [reply]
Version 3.20.0, which includes this new feature, is now released

**Reply by francwalter** [reply]
> Version 3.20.0, which includes this new feature, is now released

Sorry, only now I realized that you developed it already. 
Thanks a lot it works flawlessly, great work! 

The feature is in: 

+-Button > Create from scratch> Response handling > Display settings > Response display > Enable JavaScript (experimental)
